hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/security/RouterDelegationTokenSecretManager.java

```diff
@@ -157,6 +184,29 @@ public void updateStoredToken(RMDelegationTokenIdentifier id, long renewDate) th
     }
   }
 
+  /**
+   * The Router Supports Update Token.
+   *
+   * @param identifier RMDelegationToken.
+   * @param tokenInfo DelegationTokenInformation.
+   */
+  public void updateStoredToken(RMDelegationTokenIdentifier identifier,
+      DelegationTokenInformation tokenInfo) {
+    try {
+      byte[] tokenInfoBytes =
+          RouterDelegationTokenSupport.encodeDelegationTokenInformation(tokenInfo);
+      long renewDate = tokenInfo.getRenewDate();
+      String token = Base64.getUrlEncoder().encodeToString(tokenInfoBytes);
```

Comment:
We always do this.
Why not make encodeDelegationTokenInformation() return the String encoded already?

Reply:
Thanks for your suggestion, I will modify this part of the code.